### PR TITLE
Fixes to Transport

### PR DESCRIPTION
**Pull request by krisread**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -80,9 +76,12 @@ module.exports = function (options) {
     tu.make_client(make_send, client_options, clientdone)
 
     function make_send (spec, topic, send_done) {
+      // Topic can be something like "pin:something" to match listener pin
+      if (args.topic) topic = 'seneca_' + args.topic.replace(':', '_') + '_';
```
Comment (krisread) [inline lib/index.js:80]: ^ This allows you to pass in a "topic" to the client.  The server chooses its topic based on what you pin but this doesn't work for the client... so have to override it like this in the transport implementation.

**Comment by balmasi** [comment]
@krisread damn it you're right. You explained this earlier :man_facepalming:

**Comment by krisread** [comment]
Yeah I thought of using "pin" as well to make it more consistent but... seems like seneca is to blame in sabotaging this plan. 

It might be possible to totally rewrite the transport somehow to not use `tu.make_client` at ALL and build the client custom in the transport layer. I think "make_client" is the real broken thing so just "not using it" might be an option?   I think that would be possible but more work!   In theory you can push a lot of logic into the transport if you try.

File: lib/index.js [hunk 2]
```diff
@@ -80,9 +76,12 @@ module.exports = function (options) {
     tu.make_client(make_send, client_options, clientdone)
```
Comment (krisread) [inline lib/index.js:76]: ^ This is the line that "breaks" when you send in args with "pins"... 

An alternative might be to not call this at all, maybe there is another way to hook make_send (below) into seneca?  Maybe copy/paste some logic from transport_utils.js

Comment (balmasi) [reply lib/index.js:76]: @krisread I think it makes sense to get this working for now
then go back and switch to/write a better transport. the AMQP one seems pretty mature, but I'd have to dig in

**Pull request merged**